apm-lambda-extension/extension/route_handlers.go

```diff
@@ -30,19 +32,30 @@ type AgentData struct {
 }
 
 var AgentDoneSignal chan struct{}
+var mainExtensionContext context.Context
 
 // URL: http://server/
-func handleInfoRequest(apmServerUrl string) func(w http.ResponseWriter, r *http.Request) {
+func handleInfoRequest(ctx context.Context, apmServerUrl string, config *extensionConfig) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 
+		Log.Debug("Handling APM Server Info Request")
+		mainExtensionContext = ctx
+
 		// Init reverse proxy
 		parsedApmServerUrl, err := url.Parse(apmServerUrl)
 		if err != nil {
 			Log.Errorf("could not parse APM server URL: %v", err)
 			return
 		}
+
 		reverseProxy := httputil.NewSingleHostReverseProxy(parsedApmServerUrl)
 
+		reverseProxyTimeout := time.Duration(config.DataForwarderTimeoutSeconds) * time.Second
+		reverseProxy.Transport = http.DefaultTransport
```

Comment:
The caller of `httputil.NewSingleHostReverseProxy(parsedApmServerUrl)` should not need to have any knowledge about internas of how the `reverseProxy` is built. With this change the whole `transport` of the `reverseProxy` is still switched out. What I originally meant, was that only the `timeout` should eb customized here, or even better, be passed into the abstracting `httputil.NewSingleHostReverseProxy` method.